packages/url-loader/src/plugins/replace-background.ts

```diff
@@ -1,58 +1,55 @@
-import { z } from "zod";
-import type { ImageOptions } from "../types/image.js";
-import type { TransformationPlugin } from "../types/plugins.js";
-
-export const replaceBackgroundProps = {
-  replaceBackground: z
-    .union([
-      z.boolean(),
-      z.string(),
-      z.object({
-        seed: z.number().optional(),
-        prompt: z.string().optional(),
-      }),
-    ])
-    .describe(
-      JSON.stringify({
-        text: "Replaces the background of an image with an AI-generated background.",
-        url: "https://cloudinary.com/documentation/transformation_reference#e_gen_background_replace",
-      })
-    )
-    .optional(),
-};
-
-export const replaceBackgroundPlugin = {
-  props: replaceBackgroundProps,
-  assetTypes: ["image", "images"],
-  plugin: (settings) => {
-    const { cldAsset, options } = settings;
+import { plugin } from "../lib/plugin.js";
+
+export declare namespace ReplaceBackgroundPlugin {
+  export interface Options {
+    /**
+     * @description Replaces the background of an image with an AI-generated background.
+     * @url https://cloudinary.com/documentation/transformation_reference#e_gen_background_replace
+     */
+    replaceBackground?: NestedOptions;
```

Comment:
@ssalbdivad just to clarify how this is supposed to look / work - replaceBackground takes both a boolean and a string as well, would this end up looking like NestedOptions | string | boolean?

Reply:
Yes you're right this was my mistake leaving those out!